app/src/main/java/de/westnordost/streetcomplete/ui/common/image_select/ImageList.kt

```diff
@@ -0,0 +1,48 @@
+package de.westnordost.streetcomplete.ui.common.image_select
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.itemsIndexed
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import de.westnordost.streetcomplete.quests.bicycle_repair_station.BicycleRepairStationService
+import de.westnordost.streetcomplete.quests.bicycle_repair_station.asItem
+
+@Composable
+fun <T> ImageList(imageItems: List<ImageListItem<T>>, onClick: (index: Int, item: ImageListItem<T>) -> Unit, modifier: Modifier = Modifier, itemsPerRow: Int = 4) {
+    LazyVerticalGrid(columns = GridCells.Fixed(itemsPerRow),
+        horizontalArrangement = Arrangement.spacedBy(4.dp), verticalArrangement = Arrangement.spacedBy(4.dp),
+        modifier = modifier.height(400.dp)
+        ) {
+        itemsIndexed(imageItems) { index, item ->
+            SelectableImageItem(
```

Comment:
Ah, of course, regarding my earlier comment about different layouts for the items:

Simply 

```kotlin
  fun ImageList(
    ...,
    itemContent: @Composable (item: Item<T>) -> Unit = { /* reference to default composable here */} )`
```

Reply:
That is v similar to what I did in the end.